File: Examples/VisualBasic/Outlook/CheckPasswordProtection.vb
Imports Aspose.Email.Outlook
Imports Aspose.Email.Outlook.Pst

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Email for .NET API reference 
'when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Email for .NET API from http://www.aspose.com/downloads, 
'install it and then add its reference to this project. For any issues, questions or suggestions 
'please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Email.Examples.VisualBasic.Email.Outlook
    Class CheckPasswordProtection
        Public Shared Sub Run()
            ' ExStart:CheckPasswordProtection
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Outlook()

            Dim mapiMessage__1 As MapiMessage = MapiMessage.FromFile(dataDir & Convert.ToString("message1.msg"))
            FollowUpManager.MarkAsCompleted(mapiMessage__1)
            mapiMessage__1.Save(dataDir & Convert.ToString("MarkedCompleted_out.msg"))
        End Sub

        ' <summary>
        ' Determines whether the specified PST is password protected.
        ' </summary>
        Private Shared Function IsPasswordProtected(pst As PersonalStorage) As Boolean
            ' ExStart:CheckPasswordProtection-IsPasswordProtected
            ' If the property exists and is nonzero, then the PST file is password protected.
            If pst.Store.Properties.Contains(MapiPropertyTag.PR_PST_PASSWORD) Then
                Dim passwordHash As Long = pst.Store.Properties(MapiPropertyTag.PR_PST_PASSWORD).GetLong()
                Return passwordHash <> 0
            End If
            Return False
            ' ExEnd:CheckPasswordProtection-IsPasswordProtected
        End Function
    End Class
End Namespace

File: Examples/CSharp/Outlook/CheckPasswordProtection.cs
using Aspose.Email.Mapi;
using Aspose.Email.Storage.Pst;
using System;
using System.Text;

/*
This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Email for .NET API reference 
when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Email for .NET API from http://www.aspose.com/downloads, 
install it and then add its reference to this project. For any issues, questions or suggestions 
please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
*/

namespace Aspose.Email.Examples.CSharp.Email.Outlook
{
    class CheckPasswordProtection
    {
        public static void Run()
        {
            // The path to the File directory.
            string dataDir = RunExamples.GetDataDir_Outlook();

            using (PersonalStorage pst = PersonalStorage.FromFile(dataDir + "passwordprotectedPST.pst"))
            {
                Console.WriteLine("PST is protected: {0}",IsPasswordProtected(pst));
            }
        }

        /// <summary>
        /// Determines whether the specified PST is password protected.
        /// </summary>
        private static bool IsPasswordProtected(PersonalStorage pst)
        {
            // If the property exists and is nonzero, then the PST file is password protected.
            if (pst.Store.Properties.ContainsKey(MapiPropertyTag.PR_PST_PASSWORD))
            {
                long passwordHash = pst.Store.Properties[MapiPropertyTag.PR_PST_PASSWORD].GetLong();
                return passwordHash != 0;
            }
            return false;
        }
    }
}